### PR TITLE
Add serde support for protobuf generated struct.

### PR DESCRIPTION
**Pull request by jmjoy**
So protobuf generated struct can be serialized and deserialized.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ futures-core = "0.3.21"
 futures-util = "0.3.21"
 prost = "0.10.4"
 prost-derive = "0.10.1"
+serde = { version = "1.0.138", features = ["derive"] }
```
Comment (jmjoy) [inline Cargo.toml:47]: There will be a new dependency [serde_derive](https://crates.io/crates/serde_derive), it's under [MIT ](https://choosealicense.com/licenses/mit)OR [Apache-2.0](https://choosealicense.com/licenses/apache-2.0) license, so it's OK.

**Review by wu-sheng — APPROVED** [review]
Curious question, how we do serialization now?

**Comment by jmjoy** [comment]
> Curious question, how we do serialization now?

The protobuf library use it's own trait [prost::Message](https://docs.rs/prost/latest/prost/trait.Message.html), not the most common serialize/deserialize library `serde`.

**Comment by wu-sheng** [comment]
I see. So, the serialization would change to use serde automatically after this change, we don't need explicitly calling the method, it is integrated into generated codes. Am I right?

**Comment by jmjoy** [comment]
> I see. So, the serialization would change to use serde automatically after this change, we don't need explicitly calling the method, it is integrated into generated codes. Am I right?

Yes, for the serialization scenes which need `serde`.

**Pull request merged**